### PR TITLE
Fix (some) HLint issues in wire-api

### PR DESCRIPTION
**Pull request by supersven**
Fixing these issues has two benefits:

- The code is (hopefully) easier to read.
- HLint issues that might indicate errors are easier to spot. (Less "noise" in HLint's output.)

## Checklist

 - [X] The **PR Title** explains the impact of the change.
 - [X] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] **changelog.d** contains the following bits of information ([details](https://github.com/wireapp/wire-server/blob/develop/docs/developer/changelog.md)):
   - [X] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.

### CODE REVIEW

File: libs/wire-api/src/Wire/API/Arbitrary.hs [hunk 1]
```diff
@@ -1,6 +1,5 @@
-{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE DisambiguateRecordFields #-}
```
Comment (fisx) [inline libs/wire-api/src/Wire/API/Arbitrary.hs:1]: oops?  why did you turn this on if it obviously compiled before?

Comment (fisx) [reply libs/wire-api/src/Wire/API/Arbitrary.hs:1]: ah, probably some overlap with RecordWildCards?

File: libs/wire-api/src/Wire/API/Conversation.hs [hunk 2]
```diff
@@ -153,11 +153,10 @@ conversationMetadataObjectSchema =
...
     <*> cnvmMessageTimer
-      .= ( optFieldWithDocModifier
-             "message_timer"
-             (description ?~ "Per-conversation message timer (can be null)")
-             (maybeWithDefault A.Null schema)
-         )
+      .= optFieldWithDocModifier
+        "message_timer"
+        (description ?~ "Per-conversation message timer (can be null)")
+        (maybeWithDefault A.Null schema)
```
Comment (fisx) [inline libs/wire-api/src/Wire/API/Conversation.hs:159]: @battermann and this is why you should not mix hlint changes and feature work in the same PR!  :)  (see #2035)

**Review by fisx — APPROVED** [review]

File: changelog.d/5-internal/fix-hlint-issues-in-wire-api [hunk 3]
```diff
@@ -0,0 +1 @@
+Fix HLint issues in libs/wire-api. There are a lot left, but these ones should be safe to not change the application's behavior.
```
Comment (pcapriotti) [inline changelog.d/5-internal/fix-hlint-issues-in-wire-api:1]: I would just write "Fix some HLint issues in wire-api". Your phrasing does not seem appropriate for the release notes, but feel free to use that text as commit message for the merge.

**Review by pcapriotti — APPROVED** [review]
Looks good.

File: libs/wire-api/src/Wire/API/Arbitrary.hs (hunk 1 above)
Comment (supersven) [reply libs/wire-api/src/Wire/API/Arbitrary.hs:1]: Indeed, `DisambiguateRecordFields` is implied by `RecordWildCards`, but `DisambiguateRecordFields` is sufficient here.

Probably, this doesn't buy us much, but reducing HLint warnings helps to make the more important ones easier to spot...

File: changelog.d/5-internal/fix-hlint-issues-in-wire-api (hunk 3 above)
Comment (supersven) [reply changelog.d/5-internal/fix-hlint-issues-in-wire-api:1]: Yes, that sounds reasonable. :+1:

Comment (supersven) [reply changelog.d/5-internal/fix-hlint-issues-in-wire-api]: (Fixed according to the comment)

**Pull request merged**